asgiref/server.py

```diff
@@ -134,7 +136,7 @@ async def application_checker(self):
         """
         while True:
             await asyncio.sleep(self.application_checker_interval)
-            for scope_id, details in list(self.application_instances.items()):
+            for scope_id, details in self.application_instances.items():
```

Comment:
Have you checked to make sure this still works under mutation? That's why the `list()` was there in the first place.

Reply:
The new change passed the testing of pytest.  I made this change just because the similar method is called without list() constructor in the previous delete_oldest_application_instance method.

```
    def delete_oldest_application_instance(self):
        """
        Finds and deletes the oldest application instance
        """
        oldest_time = min(
            details["last_used"] for details in self.application_instances.values()
        )
        for scope_id, details in self.application_instances.items():
            if details["last_used"] == oldest_time:
                self.delete_application_instance(scope_id)
                # Return to make sure we only delete one in case two have
                # the same oldest time
                return
```

Reply:
Oh, sorry. I find that list() constructor is meant to avoid dictionary changed size during iteration.